src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -579,6 +609,16 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
                 out.writeVInt(INLINE_VAL); //Default to inline
             }
         }
+
+        @Override
+        public String toString() {
+            //ScriptModes looks for 'dynamic' rather than 'inline' when reading settings, that's the terminology
+            //we always used with users, although we prefer inline internally
+            if (this == INLINE) {
```

Comment:
+1 on @s1monw's comment... though I'd still reconsider changing it in the rest layer... if we change it to the user, this change should be reflected in the code as well (e.g. java clients)... I'd keep it "inline" everywhere for now...